File: packages/shared/test/native-status.test.mjs
import {readFileSync} from 'node:fs';
import test from 'node:test';
import assert from 'node:assert/strict';
import {nativeStatusView} from '../dist/native-status.js';
const fixture=JSON.parse(readFileSync(new URL('../../../adapters/ui/fixtures/native-status.json',import.meta.url),'utf8'));
for(const row of fixture)test(row.name,()=>assert.deepEqual(nativeStatusView(row.facts),row.expected));

File: apps/android/app/src/test/java/dev/mote/collector/NativeStatusTest.kt
package dev.mote.collector
import org.json.JSONArray
import org.json.JSONObject
import org.junit.Assert.*
import org.junit.Test
class NativeStatusTest {
    private fun value(input: Any?): Any? = when(input) {
        is JSONObject -> input.keys().asSequence().associateWith { value(input.get(it)) }
        is JSONArray -> (0 until input.length()).map { value(input.get(it)) }
        is Number -> input.toDouble()
        JSONObject.NULL -> null
        else -> input
    }
    @Test fun sharedArchiveProcessingAndSourceContract() {
        val rows = JSONArray(javaClass.classLoader!!.getResource("native-status.json")!!.readText())
        for (i in 0 until rows.length()) {
            val row = rows.getJSONObject(i)
            assertEquals(row.getString("name"), value(row.getJSONObject("expected")), value(NativeStatus.project(row.getJSONObject("facts"))))
        }
    }
}
